src/libs/PolicyUtils.ts

```diff
@@ -0,0 +1,217 @@
+    }
+    return numValue.toFixed(CONST.CUSTOM_UNITS.RATE_DECIMALS);
+}
+
+function getRateDisplayValue(value: number, toLocaleDigit: (arg: string) => string): string {
+    const numValue = getNumericValue(value, toLocaleDigit);
+    if (Number.isNaN(numValue)) {
+        return '';
+    }
+    return numValue.toString().replace('.', toLocaleDigit('.')).substring(0, value.toString().length);
+}
+
+function getUnitRateValue(customUnitRate: UnitRate, toLocaleDigit: (arg: string) => string) {
+    return getRateDisplayValue((customUnitRate?.rate ?? 0) / CONST.POLICY.CUSTOM_UNIT_RATE_BASE_OFFSET, toLocaleDigit);
+}
+
+/**
+ * Get the brick road indicator status for a policy. The policy has an error status if there is a policy member error, a custom unit error or a field error.
+ */
+function getPolicyBrickRoadIndicatorStatus(policy: OnyxTypes.Policy, policyMembersCollection: PolicyMembersCollection): string {
+    const policyMembers = policyMembersCollection?.[`${ONYXKEYS.COLLECTION.POLICY_MEMBERS}${policy.id}`] ?? {};
+    if (hasPolicyMemberError(policyMembers) || hasCustomUnitsError(policy) || hasPolicyErrorFields(policy)) {
+        return CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR;
+    }
+    return '';
+}
+
+/**
+ * Check if the policy can be displayed
+ * If offline, always show the policy pending deletion.
+ * If online, show the policy pending deletion only if there is an error.
+ * Note: Using a local ONYXKEYS.NETWORK subscription will cause a delay in
+ * updating the screen. Passing the offline status from the component.
+ */
+function shouldShowPolicy(policy: OnyxTypes.Policy, isOffline: boolean): boolean {
+    return (
+        policy &&
+        policy?.isPolicyExpenseChatEnabled &&
+        policy?.role === CONST.POLICY.ROLE.ADMIN &&
+        (isOffline || policy?.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE || Object.keys(policy.errors ?? {}).length > 0)
+    );
+}
+
+function isExpensifyTeam(email: string): boolean {
+    const emailDomain = Str.extractEmailDomain(email ?? '');
+    return emailDomain === CONST.EXPENSIFY_PARTNER_NAME || emailDomain === CONST.EMAIL.GUIDES_DOMAIN;
+}
+
+function isExpensifyGuideTeam(email: string): boolean {
+    const emailDomain = Str.extractEmailDomain(email ?? '');
+    return emailDomain === CONST.EMAIL.GUIDES_DOMAIN;
+}
+
+/**
+ * Checks if the current user is an admin of the policy.
+ */
+const isPolicyAdmin = (policy: OnyxTypes.Policy): boolean => policy?.role === CONST.POLICY.ROLE.ADMIN;
+
+/**
+ * Create an object mapping member emails to their accountIDs. Filter for members without errors, and get the login email from the personalDetail object using the accountID.
+ *
+ * We only return members without errors. Otherwise, the members with errors would immediately be removed before the user has a chance to read the error.
+ */
+function getMemberAccountIDsForWorkspace(policyMembers: PolicyMemberList, personalDetails: PersonalDetailsList): MemberEmailsToAccountIDs {
```

Comment:
```suggestion
function getMemberAccountIDsForWorkspace(policyMembers: OnyxEntry<OnyxTypes.PolicyMembers>, personalDetails: OnyxEntry<PersonalDetailsList>): MemberEmailsToAccountIDs {
```

`personalDetails` can be `null`, context [here](https://github.com/Expensify/App/blob/main/src/pages/workspace/WorkspaceInvitePage.js#L71) and [here](https://github.com/Expensify/App/blob/main/src/pages/workspace/WorkspaceInvitePage.js#L261).